README.md

```diff
@@ -17,42 +17,85 @@ If you rely heavily on mobile interactions and are looking for a mobile-first UI
+
+These private packages define development dependencies and contain build configuration. They adhere to standard NPM package layout &mdash; this allows us to keep clear API boundaries for build configuration and isolate groups of `devDependencies`. In the future, we might publish these packages to allow other Blueprint projects to use this infrastructure outside this monorepo.
+
+- `karma-build-scripts`
+- `node-build-scripts`
+- `tslint-config`
+- `webpack-build-scripts`
 
 ## Development
 
-We use [Lerna](https://lernajs.io/) to manage inter-package dependencies in this monorepo.
-Builds are orchestrated via [Gulp](http://gulpjs.com/) tasks.
+[Lerna](https://lernajs.io/) manages inter-package dependencies in this monorepo.
+Builds are orchestrated via `lerna run` and NPM scripts.
 
 __Prerequisites__: Node.js v8+, Yarn v1.0+
 
-1. `git clone` this repository (or fork if you lack permissions).
-1. `yarn` to install dependencies at the root of the repo.
-1. `yarn bootstrap` to install package dependencies install and symlink them using [Lerna](https://lernajs.io/).
-1. `yarn gulp` to compile and start the server and watcher.
-1. Open your browser to [localhost:9000/packages/site-docs/dist/](http://localhost:9000/packages/site-docs/dist/).
+### One-time setup
+
+After cloning this repo, run:
+
+1. `yarn` to install all dependencies.
+1. `yarn verify` to ensure you have all the build tooling working properly.
+
+### Incorporating upstream changes
+
+If you were previously in a working state and have just pulled new code from `develop`:
+
+- If there were package dependency changes, run `yarn` at the root.
+  - This command is very quick if there are no new things to install.
+- Run `yarn compile-libs` to get the latest built versions of the library packages in this repo.
+  - This command is quicker than `yarn verify` since it doesn't build the application packages (`workspace-chrome`, `workspace-docs`, etc.) or run tests
 
-### Updating build dependencies
+### Developing libraries
 
-1. `yarn add <package-name>` or `yarn upgrade <package-name>`.
+Each library has its own dev script which you can run to watch changes to that package and run the docs application with webpack-dev-server: `yarn dev:core`, `yarn dev:datetime`, etc.
+
+- One exception is table &mdash; since it has its own playground application for development, it `dev:table` script doesn't run the docs site.
+- You may also choose to watch changes across all packages with `dev:all`.
+
+### Updating dependencies
+
+1. Edit the `package.json` where you wish to change dependencies.
+1. Run `yarn` at the root to update lockfiles.
 1. Commit the result.
 
-### Updating package dependencies
+### Updating documentation
+
+Much of Blueprint's documentation lives inside source code as JSDoc comments in `.tsx?` files and KSS markup in `.scss` files. This documentation is extracted
+and converted into static JSON data using [documentalist](https://github.com/palantir/documentalist/).
```

Comment:
nit: newlines here are kinda weird. can we kick it to one line for simplicity in the editor?